OpenRA.Mods.Common/Traits/Render/RenderDetectionCircle.cs

```diff
@@ -71,7 +74,7 @@ IEnumerable<IRenderable> RenderCircle(Actor self, DetectionCircleVisibility visi
 				yield break;
 
 			yield return new DetectionCircleAnnotationRenderable(
-				self.CenterPosition,
+				self.CenterPosition - new WVec(WDist.Zero, WDist.Zero, info.RenderOnGround ? self.World.Map.DistanceAboveTerrain(self.CenterPosition) : WDist.Zero),
```

Comment:
```suggestion
				position,
```

And then above add
```csharp
var position = self.CenterPosition;
if (info.RenderOnGround)
	position -= new WVec(0, 0, self.World.Map.DistanceAboveTerrain(position).Length);
```

This should be better style

Reply:
Fixed